File: scripts/test-auth.js
'use strict';
const assert = require('assert');

// test parseTokenResponse
let { parseTokenResponse, buildCredentialsJson } = require('./ecc-auth');

const raw = 'access_token=gho_abc&token_type=bearer&scope=read%3Auser';
const parsed = parseTokenResponse(raw);
assert.strictEqual(parsed.access_token, 'gho_abc');
assert.strictEqual(parsed.token_type, 'bearer');
assert.strictEqual(parsed.scope, 'read:user');
console.log('✓ parseTokenResponse');

// test buildCredentialsJson
const creds = buildCredentialsJson({ access_token: 'gho_abc', token_type: 'bearer', scope: 'read:user' }, 'testuser');
const obj = JSON.parse(creds);
assert.strictEqual(obj.access_token, 'gho_abc');
assert.strictEqual(obj.github_login, 'testuser');
assert.strictEqual(obj.token_type, 'bearer');
assert.ok(obj.stored_at, 'stored_at should be set');

// verify stored_at is valid ISO date
assert.ok(!isNaN(Date.parse(obj.stored_at)), 'stored_at should be valid date');
console.log('✓ buildCredentialsJson');

console.log('All tests passed');

File: tests/test_auth.py
import json
import pytest
from pathlib import Path
from unittest.mock import patch
from llm.core.auth import load_credentials, CredentialsNotFoundError, CredentialsInvalidError
from llm.providers.claude import ClaudeProvider
from llm.providers.openai import OpenAIProvider
from llm.providers.ollama import OllamaProvider

def test_load_credentials_missing_file(tmp_path):
    with patch("llm.core.auth.CREDENTIALS_PATH", tmp_path / "credentials.json"):
        with pytest.raises(CredentialsNotFoundError):
            load_credentials()

def test_load_credentials_missing_token(tmp_path):
    creds_file = tmp_path / "credentials.json"
    creds_file.write_text(json.dumps({"token_type": "bearer"}))
    with patch("llm.core.auth.CREDENTIALS_PATH", creds_file):
        with pytest.raises(CredentialsInvalidError):
            load_credentials()

def test_load_credentials_valid(tmp_path):
    creds_file = tmp_path / "credentials.json"
    creds = {"access_token": "gho_abc123", "token_type": "bearer", "scope": "read:user", "github_login": "testuser", "stored_at": "2026-01-01T00:00:00Z"}
    creds_file.write_text(json.dumps(creds))
    with patch("llm.core.auth.CREDENTIALS_PATH", creds_file):
        result = load_credentials()
    assert result["access_token"] == "gho_abc123"
    assert result["github_login"] == "testuser"


def test_claude_provider_requires_credentials(tmp_path):
    with patch("llm.core.auth.CREDENTIALS_PATH", tmp_path / "credentials.json"):
        with pytest.raises(CredentialsNotFoundError):
            ClaudeProvider(api_key="sk-test")


def test_openai_provider_requires_credentials(tmp_path):
    with patch("llm.core.auth.CREDENTIALS_PATH", tmp_path / "credentials.json"):
        with pytest.raises(CredentialsNotFoundError):
            OpenAIProvider(api_key="sk-test")


def test_ollama_provider_requires_credentials(tmp_path):
    with patch("llm.core.auth.CREDENTIALS_PATH", tmp_path / "credentials.json"):
        with pytest.raises(CredentialsNotFoundError):
            OllamaProvider()
